blazingcache-core/src/main/java/blazingcache/client/CacheClient.java

```diff
@@ -1693,4 +1694,24 @@ public CacheClientStatisticsMXBean getStatistics() {
         return (T) entry.resolveReference(entrySerializer);
     }
 
+    private void locallyLockKey(RawString key) {
+        synchronized (lockedKeys) {
```

Comment:
aren't we serialising all the operations of the client ? we are not locking "per-key" here

IIRC in BlazingCache we already have good utilities for per-key locking, at least in the server

Reply:
It it serializing the whole server but just for the access at the lockedKeys set. Waiting threads won't block other to enter the serialization block and they will block only if can't add an element in lockedKeys

Reply:
Idk if it's worth it. Even with that we will still serialize the lock creation/retrieval since all operations are guarded by a general lock